### PR TITLE
[ELBOW-13] add verifyUser functionality

### PR DESCRIPTION
**Pull request by mastfissh**
Adds verifyUser() - this checks with svc-auth to get the currently logged in users details. From there, you can inspect req.user to determine roles, email, and so forth.

### CODE REVIEW

File: bin/auth-test-setup.js [hunk 1]
```diff
@@ -100,10 +100,15 @@ function groupMatching(groups){
   });
 }
 
-function addUsersToGroup(users, group){
+function addUsersToGroup(users, groups){
+  let group = groups[0]
```
Comment (forkfork) [inline bin/auth-test-setup.js]: Super minor ; comment.

File: index.js [hunk 2]
```diff
@@ -51,6 +53,22 @@ function checkAuth(){
   }
 }
 
-Auth.checkAuth = checkAuth
+function verifyUser(){
+  return function(req, res, next){
+    checkLoggedIn(req, res, function(){
+      fetch(process.env.AUTH_HOST + "/api/users/current", { headers: {Cookie: req.headers.cookie} })
+        .then(res => res.json())
+        .then(function (data) {
```
Comment (forkfork) [inline index.js]: When the access_token is bounced (4xx) we'll get a rather ugly error here I think. req.user will then contain an error response.

**Comment by forkfork** [comment]
Other than the fetch error handling comment, :+1:

File: index.js (hunk 2 above)
Comment (mastfissh) [inline index.js]: Changed it to return a null if the status code isn't ok, so user will be null if the server doesn't give back a nice response.

**Comment by forkfork** [comment]
Random gifs for Github approval from now.

very nice

**Pull request merged**